taca/analysis/analysis.py

```diff
@@ -155,6 +155,19 @@ def _upload_to_statusdb(run):
         parser.obj["DemultiplexConfig"] = {
             "Setup": {"Software": run.CONFIG.get("bcl2fastq", {})}
         }
+    # Keep PDC archive date if there is one already
+    run_vals = run.split("_")
+    if len(run_vals[0]) == 8:
+        run_date = run_vals[0][2:]
+    else:
+        run_date = run_vals[0]
+    run_fc = f"{run_date}_{run_vals[-1]}"
+    fc_names = {e.key: e.id for e in db.view("names/name", reduce=False)}
```

Comment:
This would fetch all flowcell IDs in the database. I think you can fetch the document you want directly with the key in just one call. I would have to look up the specific syntax though.

Reply:
Hi, I'm re-using the code here: https://github.com/SciLifeLab/TACA/blob/c08d3f3ca4ed2f0f8ef7e669b6d7313b746e33f0/taca/backup/backup.py#L258. It would be nice if there is some better way!

Reply:
`doc = db.view("names/name", reduce=False, include_docs=True)[run_fc].rows[0].doc`

should do the trick, but one would need to wrap it in try except to catch the case when the flowcell is not present in the view. Alternatively one could have multiple lines and check that the rows are not empty after fetching.